spot_wrapper/wrapper.py

```diff
@@ -601,90 +1030,132 @@ def _robot_command(self, command_proto, end_time_secs=None, timesync_endpoint=No
+        Executes a stand command, but one where the robot will assume the pose specified by the given parameters.
+
+        If no parameters are given this behaves just as a normal stand command
+
+        Args:
+            monitor_command: Track the state of the command in the async idle, which sets is_standing
+            body_height: Offset of the body relative to normal stand height, in metres
+            body_yaw: Yaw of the body in radians
+            body_pitch: Pitch of the body in radians
+            body_roll: Roll of the body in radians
+
+        """
+        if any([body_height, body_yaw, body_pitch, body_roll]):
+            # If any of the orientation parameters are nonzero use them to pose the body
+            body_orientation = EulerZXY(yaw=body_yaw, pitch=body_pitch, roll=body_roll)
+            response = self._robot_command(
+                RobotCommandBuilder.synchro_stand_command(
+                    body_height=body_height, footprint_R_body=body_orientation
+                )
+            )
+        else:
+            # Otherwise just use the mobility params
+            response = self._robot_command(
+                RobotCommandBuilder.synchro_stand_command(params=self._mobility_params)
+            )
+
+        if monitor_command:
+            self._last_stand_command = response[2]
+        return response[0], response[1]
+
+    @try_claim(power_on=True)
+    def battery_change_pose(self, dir_hint: int = 1):
+        """
+        Put the robot into the battery change pose
+
+        Args:
+            dir_hint: 1 rolls to the right side of the robot, 2 to the left
+        """
         if self._is_sitting:
             response = self._robot_command(
-                RobotCommandBuilder.battery_change_pose_command(1))  # 1: Rightside
+                RobotCommandBuilder.battery_change_pose_command(dir_hint)
+            )
             return response[0], response[1]
-        return False, 'Call /sit first'
+        return False, "Call sit before trying to roll over"
 
+    @try_claim
     def safe_power_off(self):
         """Stop the robot's motion and sit if possible.  Once sitting, disable motor power."""
-        response = self.claim()
-        if not response[0]:
-            return response
         response = self._robot_command(RobotCommandBuilder.safe_power_off_command())
         return response[0], response[1]
 
     def clear_behavior_fault(self, id):
         """Clear the behavior fault defined by id."""
         try:
-            rid = self._robot_command_client.clear_behavior_fault(behavior_fault_id=id, lease=None)
+            rid = self._robot_command_client.clear_behavior_fault(
+                behavior_fault_id=id, lease=None
+            )
             return True, "Success", rid
         except Exception as e:
             return False, str(e), None
 
-    def power_on(self, allow_already_powered=False):
+    @try_claim
+    def power_on(self):
```

Comment:
This removes the `allow_already_powered_on` flag in favor of always returning true if the robot was already powered on.  We weren't using the functionality of returning false if the robot was already powered on but heads up to @METEORITENMAX

Reply:
Yes, the reason I removed this was because it didn't appear to be used anywhere in the ros2 driver.

Reply:
Oh yeah I think I added that when I updated some of the power on stuff to preserve old behavior.  Since no one has needed it in the months since I did that I'm going to say it's fine to remove :)